src/lib/components/Canvas.svelte

```diff
@@ -0,0 +1,157 @@
+<script context="module" lang="ts">
+  import RenderManager from '../util/renderManager';
+  import { getContext as getCTX } from 'svelte';
+
+  export const KEY = Symbol();
+
+  interface TypedContext {
+    register: RenderManager['register'];
+    unregister: RenderManager['unregister'];
+    redraw: RenderManager['redraw'];
+  }
+
+  export const getTypedContext = (): TypedContext => getCTX(KEY);
+</script>
+
+<script lang="ts">
+  import { onMount, onDestroy, setContext } from 'svelte';
+
+  export let width = 640,
+    height = 640,
+    pixelRatio: number | null = null,
+    style = '',
+    autoclear = true;
+
+  /** Class field. Only works for global classes. */
+  let clazz = '';
+
+  export { clazz as class, redraw, getCanvas, getContext };
+
+  let canvas: HTMLCanvasElement;
+  let context: CanvasRenderingContext2D | null = null;
+  let animationLoop: number;
+  let layerRef: HTMLDivElement;
+  let layerObserver: MutationObserver;
+
+  const manager = new RenderManager();
+
+  function redraw() {
+    manager.redraw();
+  }
+
+  function getCanvas() {
+    return canvas;
+  }
+
+  function getContext() {
+    return context;
+  }
+
+  if (pixelRatio === undefined) {
+    pixelRatio = 2;
+  }
+
+  function draw() {
+    if (!pixelRatio) return;
+    if (!context) return;
```

Comment:
These checks shouldn't be necessary — we can guarantee these will be set by the time `draw` is called.